### DISCUSSION TITLE
support reset data to undefined if error ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [TanStack/query#5068] [Q&A]
```jsx

const {data,error} = useQuery()

if(error){

data = undefined // It is difficult to set the value of data. Does useQuery provide a way to reset it?
}
```

**Comment by TkDodo** [comment]
What problem are you trying to solve? I'm pretty sure there are ways to solve the problem you're having, but you're not telling the problem. You're asking for a specific implementation detail.

For what it's worth, nothing stops you from doing:

```
const query = useQuery()
const data = query.error ? undefined : query.data
```
